### PR TITLE
revolver: init at 0.2.3

### LINKED ISSUE #263959: Package request: revolver
**Project description**
A progress spinner for ZSH scripts 

**Metadata**

* homepage URL: https://github.com/molovo/revolver
* source URL: https://github.com/molovo/revolver
* license: mit
* platforms: unix

### PR DESCRIPTION
**Pull request by jfvillablanca**
## Description of changes
Resolves https://github.com/NixOS/nixpkgs/issues/263959

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/revolver/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    inherit pname version;
+
+    src = fetchFromGitHub {
+      owner = "molovo";
+      repo = "revolver";
+      rev = "v${version}";
+      hash = "sha256-ACYoMERjn/uv906mffNF1qu2j08nktqQFgwrfXdZX4g=";
+    };
+
+    preInstall = "mkdir -p $out/bin";
```
Comment (eclairevoyant) [inline pkgs/development/tools/revolver/default.nix:20]:
```suggestion
```

File: pkgs/development/tools/revolver/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    installFlags = [ "PREFIX=$(out)" ];
+
+    installPhase = ''
+      runHook preInstall
+
+      cp revolver $out/bin/revolver
```
Comment (eclairevoyant) [inline pkgs/development/tools/revolver/default.nix:27]:
```suggestion
      install -D555 revolver -t $out/bin
```

File: pkgs/development/tools/revolver/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      homepage = "https://github.com/molovo/revolver";
+      description = "A progress spinner for ZSH scripts";
+      license = licenses.mit;
+      maintainers = with maintainers; [ jfvillablanca ];
```
Comment (eclairevoyant) [inline pkgs/development/tools/revolver/default.nix:36]: please set `meta.mainProgram` and `meta.platforms`

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -42098,4 +42098,6 @@ with pkgs;
   code-maat = callPackage ../development/tools/code-maat {};
 
   mdhtml = callPackage ../tools/text/mdhtml { };
+
+  revolver = callPackage ../development/tools/revolver { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:42102]: please use `pkgs/by-name` and remove this line

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jfvillablanca) [reply pkgs/top-level/all-packages.nix:42102]: i figured it was used like a library by other scripts so I placed it under pkgs/development/tools

**Comment by d-brasher** [comment]
`revolver` was added with https://github.com/NixOS/nixpkgs/pull/327424, I think this can be closed

**Pull request closed without merging**